connexion/exceptions.py

```diff
@@ -47,20 +48,9 @@ def __repr__(self):  # pragma: no cover
         return '<ResolverError: {}>'.format(self.reason)
 
 
-class InvalidSpecification(ConnexionException):
-    def __init__(self, reason='Unknown Reason'):
-        """
-        :param reason: Reason why the specification is invalid
-        :type reason: str
-        """
-        self.reason = reason
-
-    def __str__(self):  # pragma: no cover
-        return '<InvalidSpecification: {}>'.format(self.reason)
-
-    def __repr__(self):  # pragma: no cover
-        return '<InvalidSpecification: {}>'.format(self.reason)
-
+class InvalidSpecification(ConnexionException, ValidationError):
+    pass
+    
```

Comment:
The methods from `ValidationError` are a lot more helpful, so I'm relying on mixing in those.

The exception prints like this:
```
Traceback (most recent call last):
  File "/home/daniel/stash/clean/connexion/venv/bin/connexion", line 11, in <module>
    load_entry_point('connexion', 'console_scripts', 'connexion')()
  File "/home/daniel/stash/clean/connexion/venv/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/daniel/stash/clean/connexion/venv/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/daniel/stash/clean/connexion/venv/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/daniel/stash/clean/connexion/venv/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/daniel/stash/clean/connexion/venv/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/daniel/stash/clean/connexion/connexion/cli.py", line 202, in run
    **api_extra_args)
  File "/home/daniel/stash/clean/connexion/connexion/apps/flask_app.py", line 54, in add_api
    api = super(FlaskApp, self).add_api(specification, **kwargs)
  File "/home/daniel/stash/clean/connexion/connexion/apps/abstract.py", line 152, in add_api
    options=api_options.as_dict())
  File "/home/daniel/stash/clean/connexion/connexion/apis/abstract.py", line 96, in __init__
    self._validate_spec(spec)
  File "/home/daniel/stash/clean/connexion/connexion/apis/abstract.py", line 148, in _validate_spec
    raise InvalidSpecification.create_from(e)
connexion.exceptions.InvalidSpecification: 'somestring' is not of type 'integer'

Failed validating 'type' in schema:
    {'default': 'somestring',
     'in': 'query',
     'name': 'foo',
     'type': 'integer'}

On instance:
    'somestring'
```